### PR TITLE
[Vue] migrate widget-loader directive

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR is based off of #18423.**

Changes:
* Migrate widgetloader directive.
* Allow sending custom headers in AjaxHelper.fetch

image

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/CoreHome/vue/src/WidgetLoader/WidgetLoader.vue [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+      const { widgetContent } = this.$refs;
+      if (widgetContent) {
+        widgetContent.innerHTML = '';
+      }
+      if (this.currentScope) {
+        this.currentScope.$destroy();
+      }
+    },
+    getWidgetUrl(parameters: Record<string, unknown>): Record<string, unknown> {
+      // TODO: test this
```
Comment (peterhashair) [inline plugins/CoreHome/vue/src/WidgetLoader/WidgetLoader.vue]: do we still want to add a test for this?

**Comment by peterhashair** [comment]
the interface worked as expected, but it seems like failed a test, not sure that is related. but it doesn't close the notification bar on the test I think.
```
 2) Notifications
       should still show persistent notifications on reload:
          the 'compare' command output could not be parsed, should be an integer, got: compare-im6.q16: image widths or heights differ `/home/travis/build/matomo-org/matomo/plugins/CoreHome/tests/UI/expected-screenshots/Notifications_reloaded.png' @ error/compare.c/CompareImageCommand/1000.
            Generated screenshot: /home/travis/build/matomo-org/matomo/tests/UI/processed-ui-screenshots/Notifications_reloaded.png
            Expected screenshot: /home/travis/build/matomo-org/matomo/plugins/CoreHome/tests/UI/expected-screenshots/Notifications_reloaded.png
       
            Url to reproduce: http://localhost/tests/PHPUnit/proxy/?module=CoreAdminHome&action=home&idSite=1&period=day&date=yesterday
            No captured console logs
```

File: plugins/CoreHome/vue/src/WidgetLoader/WidgetLoader.vue (hunk 1 above)
Comment (diosmosis) [reply plugins/CoreHome/vue/src/WidgetLoader/WidgetLoader.vue]: thanks for catching that, this was a reminder to do a manual test. I'll do that and remove the comment.

**Comment by diosmosis** [comment]
UI tests should be passing for this PR

**Comment by peterhashair** [comment]
there is a test that failed, I think it's related to this. Maybe need to update the tests.

File: plugins/CoreHome/stylesheets/layout.less [hunk 2]
```diff
@@ -190,7 +190,7 @@ nav {
       padding-bottom: 20px;
 
       .menu-icon {
-        padding-right: 13px;
+        padding-right: 9.5px;
```
Comment (sgiehl) [inline plugins/CoreHome/stylesheets/layout.less]: Guess this got reverted due to the merge:
```suggestion
        padding-right: 13px;
```

**Review by sgiehl — COMMENTED** [review]
Had a look through the code and did some basic local testing. Besides the one comment I already left, I guess it looks good so far. Guess some tests / expected files will need an update

**Review by sgiehl — APPROVED** [review]
pushed another small fix to fix the js tests and updated the expected screenshot files. Green build 🎉

**Pull request merged**